include/qpid/dispatch/ctools.h

```diff
@@ -217,4 +218,41 @@ do {                                    \
 #define MIN(a,b) (((a)<(b))?(a):(b))
 #define MAX(a,b) (((a)>(b))?(a):(b))
 
+//
+// Heap allocation with abort() on failure
+//
+
+static inline void *qd_malloc(size_t size)
```

Comment:
can it ever happen in dispatch that size==0 in calling this?

Reply:
Ah - good point.   I'm pretty sure there's no zero size use cases, but I'll add an assert() to enforce that contract.
If size == 0 the caller shouldn't make the call at all, or fall back to using malloc directly if they're doing something... weird.